chore: upgrade eth-lattice-keyring to v1.1.0

yilmazbahadir opened the pull request:
## **Description**

Upgrades `eth-lattice-keyring` from v0.12.4 (patched) to v1.1.0 for Lattice hardware wallet support.

**Key changes:**
- Upgraded `eth-lattice-keyring` to v1.1.0, which brings in `gridplus-sdk@4.0.0` with modern dependencies (`ox`, `abitype`, `@noble/*`, `viem`)
- Created a patch for `gridplus-sdk@4.0.0` to remove `process.env.ETHERSCAN_KEY` references that would break in browser context due to runtime process blocking
- Added Lattice-related packages (`eth-lattice-keyring`, `gridplus-sdk`, `viem`) to browserify babelify transpilation to handle modern ES syntax (e.g., `??=` operator)
- Updated LavaMoat browserify policy overrides for new package dependencies (`ox`, `@noble/curves`, `abitype`)
- Updated build-system LavaMoat policy for `tsx>fsevents`
- Regenerated all LavaMoat browserify policies (main/beta/flask/experimental)

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/38987?quickstart=1)

## **Changelog**

CHANGELOG entry: null

## **Related issues**

Fixes:

## **Manual testing steps**

1. Build the extension (`yarn dist`)
2. Load the extension in browser
3. Go to Settings > Hardware Wallets
4. Connect a Lattice hardware wallet
5. Verify that wallet connection and signing operations work correctly

## **Screenshots/Recordings**

### **Before**

N/A - Dependency upgrade, no UI changes

### **After**

Dependency upgrade, no UI changes, here is a basic smoke test:

https://www.loom.com/share/7813db9998e74653a95382e63838b21e?focus_title=1&muted=1&from_recorder=1

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [ ] I've included tests if applicable
- [x] I've documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I've applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Upgrades Lattice hardware wallet stack and build/policy plumbing to support newer dependencies.
> 
> - Bumps `eth-lattice-keyring` to `^1.1.0` bringing `gridplus-sdk@4.0.0`; adds Yarn patch to `gridplus-sdk` to remove `process.env.ETHERSCAN_KEY` usage in `dist/index.{cjs,mjs}` and updates its deps in lockfile
> - Extends Browserify `babelify` transpilation list for modern syntax in `eth-lattice-keyring`, `gridplus-sdk`, `viem`, `ox`, `uuid`, `isows`
> - Regenerates and expands LavaMoat Browserify policies (main/beta/flask/experimental + override) for new packages/globals (`viem`, `ox`, `@noble/*`, `abitype`, `bs58{,check}`, `cbor{,-bigdecimal}`, `secp256k1`, `zod`, etc.) and removes obsolete entries
> - Adds deprecation filter for `@types/uuid` in `.yarnrc.yml`; updates `package.json` resolutions to map `gridplus-sdk@4.0.0` patch and switches `eth-lattice-keyring` from patched `0.12.4` to `^1.1.0`
> - Minor TS hygiene: use `import type { Hex }` from `viem` in a few files
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit b2ee45866a7fcfccff906497dbe764875c5cc2db. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

yilmazbahadir committed:
chore: upgrade eth-lattice-keyring to v1.1.0

- Patch gridplus-sdk@4 to avoid process.env.ETHERSCAN_KEY (keeps runtime process blocked)

- Transpile eth-lattice-keyring/gridplus-sdk/viem via browserify babelify to handle modern syntax

- Regenerate LavaMoat browserify policies (main/beta/flask/experimental) for new ox/abitype/@noble/* edges

- Update build-system LavaMoat policy for tsx>fsevents

ccharly commented on package.json:
```diff
@@ -419,7 +420,7 @@
     "eth-chainlist": "~0.0.498",
     "eth-ens-namehash": "^2.0.8",
-    "eth-lattice-keyring": "patch:eth-lattice-keyring@npm%3A0.12.4#~/.yarn/patches/eth-lattice-keyring-npm-0.12.4-c5fb3fcf54.patch",
+    "eth-lattice-keyring": "1.1.0",
```
We usually use `^version`, could you change that?

```suggestion
    "eth-lattice-keyring": "^1.1.0",
```

yilmazbahadir committed:
chore: use semver caret range for eth-lattice-keyring

yilmazbahadir committed:
chore: yarn dedupe

yilmazbahadir committed:
fix: remove @types/uuid from gridplus-sdk patch

yilmazbahadir committed:
fix(build): transpile uuid and isows in browserify build

yilmazbahadir committed:
fix(build): transpile ox deps and allow abitype

yilmazbahadir committed:
chore(types): use type-only viem imports

yilmazbahadir committed:
revert(build): drop tsx fsevents policy allowlist,

yilmazbahadir committed:
chore: remove EOF newline in build-system policy

yilmazbahadir committed:
fix: correct gridplus-sdk patch hunk header

Yarn was failing to apply the gridplus-sdk patch due to a mismatched
hunk header (7/6 vs 8/7). Align the header with the actual hunk size
to restore yarn install.

yilmazbahadir committed:
fix: update gridplus-sdk patch metadata

Remove @types/uuid from the patched gridplus-sdk manifest and ensure
the patch applies cleanly so yarn install/audit reflects it.

yilmazbahadir committed:
chore: ignore @types/uuid deprecation in yarn audit

Add @types/uuid (deprecation) to npmAuditIgnoreAdvisories to silence
the audit noise coming from gridplus-sdk.
